### PR TITLE
Added Color Property in Hexdisplay, SevenSegDisplay, VariableLed, SixteenSegDisplay

### LINKED ISSUE #49: Color setting for 7-segment displays
The 7-segment and hex displays are lacking a color setting, which would be great to have. The variable LED is also lacking a color setting.

### PR DESCRIPTION
**Pull request by pulkit-30**
Fixes #49 

#### Describe the changes you have made in this PR -
Added Color Property in Hexdisplay, SevenSegDisplay, VariableLed, SixteenSegDisplay.

### Screenshots of the changes (If any) -

https://user-images.githubusercontent.com/76155456/147390616-11d6a0c8-f687-4c6d-b17e-11f293fc9534.mov

Kindly provide me with appropriate feedback so that I can continue working on this.

### CODE REVIEW

**Comment by vedant-jain03** [comment]
@pulkit-30 @nitin10s Can we add a color palette input as well that will provide easiness to the user to select a color.

**Comment by pulkit-30** [comment]
> @pulkit-30 @nitin10s Can we add a color palette input as well that will provide easiness to the user to select a color.

Sure @vedant-jain03, great Idea 💡
But I think, we need to first think and work on the code optimization, once the code is optimized and clean we can work on this as well ;)

**Comment by pulkit-30** [comment]
Hey @tachyons @gr455, Please review this PR.. and Kindly provide me with appropriate feedback so that I can continue working on this.

**Comment by gr455** [comment]
> Hey @tachyons @gr455, Please review this PR.. and Kindly provide me with appropriate feedback so that I can continue working on this.

I will review this in a couple of days. Sorry for the delay

**Comment by pulkit-30** [comment]
Hey @gr455, Please review

File: simulator/src/modules/SevenSegDisplay.js [hunk 1]
```diff
@@ -118,16 +138,16 @@ export default class SevenSegDisplay extends CircuitElement {
...
+        this.subcircuitDrawSegment(10, -17, 10, 1, ["lightgrey", this.actualColor][this.c.value], xx, yy);
+        this.subcircuitDrawSegment(-10, -20, -10, -38, ["lightgrey", this.actualColor][this.f.value], xx, yy);
+        this.subcircuitDrawSegment(-10, -17, -10, 1, ["lightgrey", this.actualColor][this.e.value], xx, yy);
+        this.subcircuitDrawSegment(-8, -38, 8, -38, ["lightgrey", this.actualColor][this.a.value], xx, yy);
+        this.subcircuitDrawSegment(-8, -18, 8, -18, ["lightgrey", this.actualColor][this.g.value], xx, yy);
+        this.subcircuitDrawSegment(-8, 1, 8, 1, ["lightgrey", this.actualColor][this.d.value], xx, yy);
 
         ctx.beginPath();
-        var dotColor = ["lightgrey", "red"][this.dot.value] || "lightgrey"
+        var dotColor = ["lightgrey", this,actualColor][this.dot.value] || "lightgrey"
```
Comment (gr455) [inline simulator/src/modules/SevenSegDisplay.js:150]: Whoops, typo here. `this.actualColor`

**Review by gr455 — CHANGES_REQUESTED** [review]
VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit)

**Comment by pulkit-30** [comment]
> This works great mostly! Thanks a lot! Extremely sorry for the late review
> 
> VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).
> 
> Also fix codeclimate, rest all looks good 👍

Great, thanks @gr455,
I will look into this and commit the changes as soon as possible .

**Comment by gr455** [comment]
> > This works great mostly! Thanks a lot! Extremely sorry for the late review
> > VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).
> > Also fix codeclimate, rest all looks good +1
> 
> Great, thanks @gr455, I will look into this and commit the changes as soon as possible .

Yep take your time

**Comment by pulkit-30** [comment]
>VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).

Hey @gr455,
May you please help me out reproducing this.

**Comment by gr455** [comment]
> > VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).
> 
> Hey @gr455, May you please help me out reproducing this.

Add a variable LED to the main circuit, go to `Edit Layout`, put drag and drop the LED onto the subcircuit

Then create a new circuit. Add Main's subcircuit to this circuit and attatch an input to the LED and see the color

**Comment by pulkit-30** [comment]
> VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).

Fixed!
May help me with this error 

Screenshot 2022-01-11 at 1 52 45 AM

**Comment by gr455** [comment]
> > VariableLED is not changing color in a subcircuit (it does change on the actual element on the parent circuit).
> 
> @gr455 , this issue is Fixed! May you help me with this error
> 
> Screenshot 2022-01-11 at 1 52 45 AM
> 
> Thanks in advance hugs

I think that's okay

**Comment by pulkit-30** [comment]
> I think that's okay

Great! Thanks

**Review by gr455 — APPROVED** [review]
Looks good to me. Thanks!

**Review by Prerna-0202 — APPROVED** [review]

**Comment by pulkit-30** [comment]
Hey @tachyons , Please check this PR.

**Pull request merged**